collect_app/src/main/java/org/odk/collect/android/external/FormUriActivity.kt

```diff
@@ -91,11 +102,62 @@ class FormUriActivity : ComponentActivity() {
             if (it != null) {
                 displayErrorDialog(it)
             } else if (savedInstanceState?.getBoolean(FORM_FILLING_ALREADY_STARTED) != true) {
-                startForm()
+                scheduler.immediate(
```

Comment:
I think we could use `Flow` here (like we do in `BlankFormListViewModel`) instead of needing to interact with `Scheduler` in an observation like this. I can imagine `FormUriViewModel` exposing a sealed result type that can be either an error, a save point or just a URI to start (instead of `FormEntryViewModel#error`).

Reply:
I've introduced such a sealed class and now the scheduler is not used in this place because reading savepoints was moved to the ViewModel.